File: workflowTradingAi/workflow.py
import time
import requests
from pprint import pprint 

URL = (
    "http://localhost:3000/api/messaging/central-channels/"
    "91b0c098-b2c3-44f8-9dbb-f0d3ca2626f4/messages"
)

payload = {
    "author_id": "f6a0a8d1-f28d-4538-bdab-5eb0b527430c",
    "content": "fais un trade",
    "server_id": "00000000-0000-0000-0000-000000000000",
    "raw_message": {
        "roomId": "91b0c098-b2c3-44f8-9dbb-f0d3ca2626f4",
        "source": "client_chat",
        "message": "fais un trade",
        "metadata": {
            "isDm": True,
            "channelType": "DM",
            "targetUserId": "49352196-f6d2-0e9b-a5cd-896a96f5119d"
        },
        "senderId": "f6a0a8d1-f28d-4538-bdab-5eb0b527430c",
        "serverId": "00000000-0000-0000-0000-000000000000",
        "channelId": "91b0c098-b2c3-44f8-9dbb-f0d3ca2626f4",
        "messageId": "00b8ecdb-fc80-4fab-a643-93586b43cf53",
        "senderName": "User-f6a0a8d1"
    },
    "metadata": {
        "isDm": True,
        "channelType": "DM",
        "targetUserId": "49352196-f6d2-0e9b-a5cd-896a96f5119d"
    },
    "source_type": "client_chat"
}

headers = {
    "Accept": "application/json",
    "Content-Type": "application/json"
}

def send_message():
    try:
        resp = requests.post(URL, json=payload, headers=headers, timeout=10)
        resp.raise_for_status()      
        print("Message créé :", resp.status_code)
        pprint(resp.json())     
    except requests.exceptions.HTTPError as e:
        print("Erreur HTTP :", e.response.status_code, e.response.text)
    except requests.exceptions.RequestException as e:
        print("Erreur réseau :", e)

while True:
    send_message()
    time.sleep(3600)

File: workflowTradingAi/workflow.ts
import cron from "node-cron";
import axios from "axios";

const BASE = "http://localhost:3000/api/messaging";

const SERVER_ZERO = "00000000-0000-0000-0000-000000000000";

const ME   = "f6a0a8d1-f28d-4538-bdab-5eb0b527430c";
const BOT  = "49352196-f6d2-0e9b-a5cd-896a96f5119d";

async function createChannel(): Promise<string> {
  const res = await axios.post(`${BASE}/central-channels`, {
    name: `btc-${Date.now()}`,
    server_id: SERVER_ZERO,
    participantCentralUserIds: [ME, BOT],
    type: "DM"
  });
  return res.data.data.id as string;
}

async function sendPrompt(roomId: string): Promise<void> {
  await axios.post(`${BASE}/central-channels/${roomId}/messages`, {
    author_id: ME,
    content: `Analyse le marché BTC en temps réel.
Calcule un sentiment compris entre 0 % et 100 %.
RENVOIE UNE SEULE LIGNE, SANS AUCUNE EXPLICATION, strictement au format :
sentiment:X%, allocation:$X, stoploss:$X, takeprofit:$X, prixBTC:$X, leverage:X

Règles :
1. Achat (bullish) → 70 % ≤ sentiment ≤ 100 %.
2. Vente (bearish) → 0 % ≤ sentiment < 30 %.
3. Attente     → 30 % ≤ sentiment ≤ 70 %.
4. Allocation ≤ 150 USDC et leverage ≤ 10 x.
5. Pour annuler une position ouverte, envoie une ligne avec un sentiment qui provoquera l’ordre inverse, même allocation & leverage.
6. Toujours adapter stop-loss et take-profit à un horizon de 15 minutes.
7. Aucune autre sortie que la ligne au format indiqué.`,
    source_type: "client_chat",
    server_id: SERVER_ZERO
  });
}

cron.schedule("*/15 * * * *", async () => {
  try {
    const id = await createChannel();
    await sendPrompt(id);
    console.log("✅ prompt envoyé dans", id);
  } catch (e: any) {
    if (e.response) {
      console.error("HTTP", e.response.status, e.response.data);
    } else {
      console.error("❌", e.message);
    }
  }
});
